docs/knowledge_base/tips/colab_with_ngrok/index.md

```diff
@@ -127,27 +138,43 @@ re-executions required. You can learn more about this in the
 Here are the new cells to add:
 
 1. Import Markdown:
-   ![Modification of the markdown](Sharing_Taipy_Ngrok_8.png){width=100%}
+   ```python
+   from taipy.gui import Gui, Markdown
+   ```
 
-2. Set empty new page:
-   ![Modification of the markdown](Sharing_Taipy_Ngrok_9.png){width=100%}
+2. Set empty new page:  
+   ```python
+   new_page=Markdown("")
+   ```
 
-3. Set content to `new_page`:
-   ![Modification of the markdown](Sharing_Taipy_Ngrok_10.png){width=100%}
+3. Set content to `new_page`: 
+   ```python
+   new_page.set_content(page)
+   ```
 
 4. Update the `pages` definition:
-   ![Modification of the markdown](Sharing_Taipy_Ngrok_11.png){width=100%}
+   ```python
+   pages = {"/":"<|toggle|theme|>\n<center>\n<|navbar|>\n</center>",
+         "line":new_page,
+         "text":page_file}
+   ```
 
 ## Variable modification with `gui.reload`
 
-1. Add this step
-   ![Variable modification](Sharing_Taipy_Ngrok_12.png){width=100%}
+1. Add this step:
+   ```python
+   gui=Gui(pages=pages)
+   ```
 
-2. Update your `tp.run(gui)`
-   ![Variable modification](Sharing_Taipy_Ngrok_13.png){width=100%}
+2. Update your `tp.run(gui)`:  
+   ```python
+   gui.run()
+   ```
 
-3. Add the `gui.reload` function
-   ![Variable modification](Sharing_Taipy_Ngrok_14.png){width=100%}
+3. Add the `gui.reload` function:
+   ```python
+   gui.reload()
```

Comment:
@FlorianJacta How can this work? gui.run() is blocking, no?

Reply:
Not in a Notebook context. Here we are on Google Colab so different cells can be executed. We could refer to the Notebook documentation now that we have it on the user man

Reply:
OK Thanks.